xapian-letor/tests/api_letor.cc

```diff
@@ -81,6 +81,79 @@ db_index_two_documents(Xapian::WritableDatabase& db, const string&)
     db.add_document(doc);
 }
 
+
+// To check for one document edge
+static void
+db_index_one_document(Xapian::WritableDatabase& db, const string&)
+{
+    Xapian::Document doc;
+    Xapian::TermGenerator termgenerator;
+    termgenerator.set_document(doc);
+    termgenerator.set_stemmer(Xapian::Stem("en"));
+    termgenerator.index_text("Tigers are solitary animals", 1, "S");
+    termgenerator.index_text("Might be that only one Tiger is good enough to "
+			     "Take out a ranker, a Tiger is a good way to "
+			     "check if a test is working or Tiger not. Tiger. "
+			     "What if the next line contains no Tigers? Would "
+			     "it make a difference to your ranker ?  Tigers  "
+			     "for the win.", 1, "XD");
+    termgenerator.index_text("The will.");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("Tigers would not be caught if one calls out the "
+			     "Tiger from the dem. Well it stopped making sens "
+			     "the best would be to stop now. But a tiger is "
+			     "something you would not like to stop. ");
+    db.add_document(doc);
+}
+
+// To check for three documents. out of which one is irrelavent
+static void
+db_index_three_documents(Xapian::WritableDatabase& db, const string&)
+{
+    Xapian::Document doc;
+    Xapian::TermGenerator termgenerator;
+    termgenerator.set_document(doc);
+    termgenerator.set_stemmer(Xapian::Stem("en"));
+    termgenerator.index_text("The will", 1, "S");
+    termgenerator.index_text("The will are considered stop words in xapian and "
+			     " would be thrown off, so the query I want to say "
+			     " is score, yes, score. The Score of a game is  "
+			     " the determining factor of a game, the score is "
+			     " what matters at the end of the day. so my advise"
+			     " to everyone is to Score it!. " , 1, "XD");
+    termgenerator.index_text(" Score might be something else too, but this para"
+			     " refers to score only at an abstract. Scores are "
+			     " in general scorring. Score it!");
+    termgenerator.increase_termpos();
+    termgenerator.index_text("Score score score is important. ");
+    db.add_document(doc);
+    doc.clear_terms();
+    termgenerator.index_text("Score score score score score score", 1, "S");
+    termgenerator.index_text("it might have an absurdly high rank in the qrel "
+			     " file or might have no rank at all in another.  "
+			     "look out for this as a testcase, as might be edgy"
+			     " Good luck and may this be with you. ", 1, "XD");
+    termgenerator.index_text(" Another irrelavent paragraph to make sure the tf"
+			     " values are down, but this increases idf values "
+			     " but let's see how this works out");
```

Comment:
Where the long strings are split above it seems random whether there's a space at the end of the first part, at the start of the second part, or both.  Functionally it doesn't matter in this case as `index_text()` doesn't care (if these were shown to the user they'd see double spaces in the "both" case, so it would matter then), but it looks messy in the source code and distracts the reader a bit.

There are also several misspellings in the sample data here.  Unless they're actually necessary for the testcase it would be better to avoid them - it makes checking for typos in messages and code comments harder.

I noticed at least "irrelavent", "scorring", "advise" (a valid word, but the wrong one - should be "advice" here), "The will are", "dem", and "sens", but there may well be others.